File: top150/ArrayString/q392.py
class Solution:
    # T: O(N), S: O(1)
    def isSubsequence(self, s, t):
        p1 = 0
        p2 = 0
        n1 = len(s)
        n2 = len(t)
        while p1 < n1 and p2 < n2:
            if s[p1] == t[p2]:
                p1 += 1
                p2 += 1
            else:
                p2 += 1
        return p1 == n1

File: top150/_02_TwoPointers/q392.java
package top150._02_TwoPointers;

public class q392 {
    public boolean isSubsequence(String s, String t) {
        int ptr = 0;

        for(char c: t.toCharArray()) {
            if(ptr == s.length()) return true;
            if(c == s.charAt(ptr)) ptr++;
        }

        return ptr == s.length();
    }
}
